n3fit/src/n3fit/hyper_optimization/filetrials.py

```diff
@@ -60,10 +64,35 @@ class FileTrials(Trials):
 
     def __init__(self, replica_path, parameters=None, **kwargs):
         self._store_trial = False
-        self._json_file = "{0}/tries.json".format(replica_path)
+        self._json_file = replica_path / "tries.json"
+        self.pkl_file = replica_path / "tries.pkl"
         self._parameters = parameters
+        self._rstate = None
         super().__init__(**kwargs)
 
+    @property
+    def rstate(self) -> Generator:
```

Comment:
Right now, the linting complains on the types as by default `self._rstate=None`. You can bypass this by importing:
```python
from typing import Union
``` 

```suggestion
    def rstate(self) -> Union[Generator, None]:
```
But personally, I'd omit any static typing here unless it is done consistently throughout the file, and it reduces the import of modules.

Reply:
I followed your suggestion and removed type hints in order to ensure consistency throughout. I added instead a more descriptive docstring. Thanks!